tests/executable_tool_bootstrap_test.py

```
@@ -0,0 +1,482 @@
+    def test_run_rejects_unsupported_fd_exec_and_nul_arguments(self) -> None:
+        bootstrapper = self.bootstrapper()
+        with mock.patch(
+            "bootstrap_executable_tools.fd_exec_supported", return_value=False
+        ):
+            with mock.patch("bootstrap_executable_tools.os.execve") as execve:
+                with self.assertRaises(BootstrapError):
+                    bootstrapper.run("fixture-a", [], {})
+        execve.assert_not_called()
+
+        with mock.patch("bootstrap_executable_tools.os.execve") as execve:
+            with self.assertRaises(BootstrapError):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)